### DISCUSSION TITLE
Opening environment doesn't work with v1.19.0

### DISCUSSION THREAD

**Discussion by RabeDeva** [mockoon/mockoon#741] [Help]
Hi all,

I have downloaded Mockoon this afternoon and following the first tutorial, I created a new environment from scratch and saved it on disk. In the new environment I then created a custom endpoint producing a simple body. Test via browser call okay, it worked. Everything perfect, easy setup, developer happy!

Then I had to close Mockoon for some reasons. Later I started again and tried to open my saved environment JSON. Nothing happens, no environment openend, the application window remains black and empty. Tried several times, tested several variants, looked for hidden save buttons or settings, no way to save and reopen an environment ... :-(

Is my intuition so wrong that when using the "Open environment" button and selecting the corresponding JSON file the formerly configured environment should reappear? Where am I going wrong? 

Unfortunately such a simple thing doesn't have a documentation entry. And the demo environment also cannot be found / opened in later starts of Mockoon for orientation. (Or should it and it's my clumsiness?)

Last observation: I also did an uninstallation of Mockoon and switched to the portable version. Starting this version the demo environment opened again and with this environment opened I was also able to open/load my custom environment. But after closing an restarting the application again: demo project not visible any more and my environment doesn't open ...

Can some explain me where's my mistake or misunderstanding please? Thank you very much!

**Comment by 255kb** [comment]
Hi @RabeDeva 
Indeed, you should be able to see your environments the next time you reload Mockoon. 
Each environment is saved in a separated JSON file. You are always prompted to choose its location when you create a new one. 

To assist you further, I would need you to check or try several things (+ sharing your OS and Mockoon's binary format you used). 
- Could you first send a screenshot of what you see when loading the application? "black and empty" is definitely not normal. 
- Also, if you could look into the application logs to see if there is any error/stack trace, it would be great (to find your app logs: https://mockoon.com/docs/latest/mockoon-data-files/settings-and-logs/)
- Finally, if you are able to create a new environment using the top left blue button (see capture below), I would be curious to see your settings file, just after the creation (to find your app settings file, same doc: https://mockoon.com/docs/latest/mockoon-data-files/settings-and-logs/)

![image](https://user-images.githubusercontent.com/7489814/176754118-e3b37425-3b0b-4500-b2d9-b04f113b9447.png)



Thank you

**Reply by RabeDeva** [reply]
Hi Guillaume,

thank's for answering! Please see the additional information below.

(1) I'm working with Windows 10 Enterprise. I have tested both Mockoon binary formats i.e. with installer and the portable version. Both behave the same way on my system.

(2) Screenshot
![mockoon-window](https://user-images.githubusercontent.com/102405965/176897014-1a72f457-11a9-485f-af71-968928294764.png)

This is the initial window I'm seeing when opening the application. Clicking the "Open environment" icon (folder symbol) a Windows File Open dialog appears but when selecting my JSON file nothing happens. The window remains in the same initial state. (And the same happens when opening the demo.json in .\Mockoon\mockoon-data\storage. I have found it now when I was looking for the logs. :-))

(3) Logfile
I'm sending you both logs I have: the installer.main.log results from my first installation, the portable.main.log from the second and current. Looks like as if the error in both is the same "[error] [APPLICATION]Unexpected error: Cannot read properties of null (reading 'storagePrettyPrint') - TypeError: Cannot read properties of null (reading 'storagePrettyPrint')"

(4) settings.json: see also the attached file.

Can you realize the error by the logs and settings?

If not and if we do not find a solution: is there maybe the option to download an older version (because I would like to continue with Mockoon - I like what I saw until now nevertheless)? And I didn't find a path to an archive at github.

Thank you & kind regards
Rabe

Ps: I had to rename the settings.json as you can see because of the github upload limitations. But it's the one I've found in C:\Tools\Mockoon\mockoon-data\storage

[installer.main.log](https://github.com/mockoon/mockoon/files/9029025/installer.main.log)
[portable.main.log](https://github.com/mockoon/mockoon/files/9029026/portable.main.log)
[settings.json.txt](https://github.com/mockoon/mockoon/files/9029032/settings.json.txt)

**Reply by RabeDeva** [reply]
Hi Guillaume,

and here also attached my own environment. As an additional observation: the application did not save all of my endpoints!
I don't know whether this was an usage error, I didn't save explicitly because there is no save button / feature available.

Best regards
Rabe
[Sst-mock1.json.txt](https://github.com/mockoon/mockoon/files/9029141/Sst-mock1.json.txt)

**Reply by 255kb** [reply]
Thank you for all the details. 
Concerning your last question, everything should be saved automatically all the time. 

I see in the logs a lot of errors, as if the app wasn't able to read the settings. 🤔
I will try to investigate further and keep you posted.

**Reply by 255kb** [reply]
I found an small issue that could cause the crash you got (I opened a new issue #751). 
If you don't want to wait until the next release you could try build the app by yourself or I could share a new binary with you so you can test (let me know).

If you want to rollback to a previous version, you can download it [here](https://github.com/mockoon/mockoon/releases/tag/v1.18.1). (But migrated environments won't open in an older version, unless you edit the file and set the `lastMigration` to 19.)
